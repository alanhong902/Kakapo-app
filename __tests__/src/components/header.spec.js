import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { Header } from 'components/';
import { getData } from '../helper';

Enzyme.configure({ adapter: new Adapter() });

function setup(props = {}) {
  const propData = {
    ...getData('themes'),
    ...getData('settings'),
    ...props,
  };
  const comp = <Header {...propData} />;
  return {
    props,
    tree: renderer.create(comp).toJSON(),
    wrapper: shallow(comp),
  };
}

test('<Header/> render', () => {
  const { tree } = setup();
  expect(tree).toMatchSnapshot();
});

test('<Header/> simulate toggle-mute click', () => {
  const onToggleMute = jest.fn();
  const { wrapper } = setup({ onToggleMute });
  wrapper.find('.toggle-mute').simulate('click');
  expect(onToggleMute).toBeCalled();
});
