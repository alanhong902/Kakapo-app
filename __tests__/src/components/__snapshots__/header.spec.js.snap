// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> render 1`] = `
<header
  className="header"
  style={
    Object {
      "backgroundColor": "#673AB7",
    }
  }
>
  <span
    className="toggle-mute"
    role="button"
    tabIndex={-1}
  >
    <span
      className="hint--right"
      data-hint="Mute"
    >
      <i
        className="icon-volume_up"
      />
    </span>
  </span>
  <div
    className="logo"
  >
    <h3
      className=""
    >
      <span
        className="logo-bg icon-img-logo"
      />
      <span
        className="logo-text"
      >
        Kakapo
      </span>
    </h3>
  </div>
</header>
`;
